File: 2thread/src/main/java/com/example/a2thread/MoonRunner.java
package com.example.a2thread;

public class MoonRunner implements Runnable{
    private long i ;
    private volatile boolean on = true;

    @Override
    public void run() {
        while(on){
            i++;
            System.out.println("i=" +i);
        }
        System.out.println("stop");
    }

    public void cancel(){
        on = false;
    }
}

File: kotlin/src/main/java/com/example/kotlin/MoonRunner.kt
package com.example.kotlin

import android.provider.Settings.Global

class MoonRunner : Runnable {
    var long : Long = 0
    override fun run() {
        while (!Thread.currentThread().isInterrupted){
            long++
            println("long = $long")
        }
        println("stop")


    }

}